Makefile

```diff
@@ -1,86 +1,50 @@
-CARGO = cargo --color always
 RUSTUP = rustup
 TOOLCHAIN = $(shell cat rust-toolchain)
-TOOLCHAIN_LOC = ""
-
-# Platform specific magic
-LIBUSB_LOC = ""
-LIBUSB_INSTALL = ""
-UNAME_S := $(shell uname -s)
-UNAME_M := $(shell uname -m)
-ifeq ($(UNAME_S),Linux)
-	LIBUSB_LOC = /usr/include/libusb-1.0
-	LIBUSB_INSTALL = sudo apt-get install --assume-yes libusb-1.0-0-dev
-	TOOLCHAIN_LOC = $(RUSTUP_DIR)/toolchains/$(TOOLCHAIN)-$(UNAME_M)-unknown-linux-gnu
-endif
-ifeq ($(UNAME_S),Darwin)
-	LIBUSB_LOC = /usr/local/lib/libusb-1.0.dylib
-	LIBUSB_INSTALL = brew install libusb
-	TOOLCHAIN_LOC = $(RUSTUP_DIR)/toolchains/$(TOOLCHAIN)-$(UNAME_M)-apple-darwin
-endif
-
-CARGO_DIR = $(HOME)/.cargo
-RUSTUP_DIR = $(HOME)/.rustup
-
-CLIPPY_LOC = $(TOOLCHAIN_LOC)/bin/cargo-clippy
-FMT_LOC = $(TOOLCHAIN_LOC)/bin/cargo-fmt
-TOMLFMT_LOC = $(CARGO_DIR)/bin/cargo-tomlfmt
+CARGO = $(RUSTUP) run --install $(TOOLCHAIN) cargo --color always
 
 build: build_debug
 
-## Build Environment
-# `test` is used to only run if the target does not exist
-# Instead of running if the pre-requisite is older than the target
-
-$(TOOLCHAIN_LOC):
-	test -e $@ || $(RUSTUP) toolchain install $(TOOLCHAIN)
-
-$(LIBUSB_LOC):
-	test -e $@ || $(LIBUSB_INSTALL)
-
-build_env: $(TOOLCHAIN_LOC) $(LIBUSB_LOC)
-
 ## Dev environment
 
-$(CLIPPY_LOC): $(TOOLCHAIN_LOC)
-	test -e $@ || $(RUSTUP) component add clippy --toolchain $(TOOLCHAIN)
+install_clippy:
+	$(RUSTUP) component list --installed --toolchain $(TOOLCHAIN) | grep -q clippy || $(RUSTUP) component add clippy --toolchain $(TOOLCHAIN)
 
-$(FMT_LOC): $(TOOLCHAIN_LOC)
-	test -e $@ || $(RUSTUP) component add rustfmt --toolchain $(TOOLCHAIN)
+install_rustfmt:
+	$(RUSTUP) component list --installed --toolchain $(TOOLCHAIN) | grep -q rustfmt || $(RUSTUP) component add rustfmt --toolchain $(TOOLCHAIN)
 
-$(TOMLFMT_LOC): $(TOOLCHAIN_LOC)
-	test -e $@ || $(CARGO) install cargo-tomlfmt
-
-dev_env: build_env $(CLIPPY_LOC) $(FMT_LOC) $(TOMLFMT_LOC)
+install_tomlfmt:
+	$(CARGO) --list |grep -q tomlfmt || $(CARGO) install cargo-tomlfmt
 
 ## User install
 
-install: build_env
-	@$(CARGO) install --path .
+install:
```

Comment:
You may already know, but this errors if the binary is already installed and asks you to pass `--force` if you want to overwrite it.

Reply:
yes, I kind of assume a first user install but you are right, I'll add it.